### ISSUE TITLE
RoBERTa issues on TPU

### ISSUE THREAD

**Issue by scheiblr** [pytorch/fairseq#2656]
Hey guys,

first of all thank you for this great open source toolkit!

RoBERTa training on TPU seems to be broken in the current release. Thus, I used an old one, which I knew worked once. Unfortunately, it failed to save the last checkpoint. Therefore, I would need to recompute starting from the last epoch's checkpoint. With the [old version](https://github.com/pytorch/fairseq/commit/c1e734b2dd7024044c8dee551620146e4f872ad4) I used, the loss is not correct, after re-triggering the training although I put the `checkpoint_last` and  `checkpoint_best` on all VMs.

The last working version (on single TPU) which I found was [this](https://github.com/pytorch/fairseq/commit/d47067937abacfe87f2963adca8daeada3c631fe).
On a TPU pod this version crashes for both: `torch-xla-nightly` and `torch-xla-1.6`. Further, I won't know if the loss would be correct in this version when continuing from a former checkpoint.

More recent versions load data, but do not start to communicate with TPUs (no XLA-compilation notification visible) and no optimizer log appears after half an hour.

I hope you guys can help, soon!
